service/controller/app/resource/chart/delete.go

```diff
@@ -2,30 +2,59 @@ package chart
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/v2/pkg/resource/crud"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 
 	"github.com/giantswarm/app-operator/v2/service/controller/app/controllercontext"
+	"github.com/giantswarm/app-operator/v2/service/controller/app/key"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
 	chart, err := toChart(deleteChange)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	if chart != nil && chart.Name != "" {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting Chart CR %#q in namespace %#q", chart.Name, chart.Namespace))
+	if key.AppName(cr) == key.ChartOperatorAppName {
+		// helm release `chart-operator` is already deleted by app-operator at this point.
+		// So app-operator should remove finalizers in the chart-operator chart CR manually.
```

Comment:
```suggestion
		// `chart-operator` helm release is already deleted by the `chartoperator` resource at this point.
		// So app-operator needs to remove finalizers so the chart-operator chart CR is deleted.
```

I was a bit confused by this until I saw it was deleted by the chartoperator resource. WDYT to this?

Reply:
Yes, that looks clearer than before. Thanks!